txbits/conf/evolutions/default/2.sql

```diff
@@ -258,12 +258,21 @@ begin
     update balances set hold = hold - new_amount
     where currency = ask.base and user_id = ask.user_id;;
 
-    update balances set hold = hold - new_amount * new_price
-    where currency = bid.counter and user_id = bid.user_id;;
+    if bid.price >  new_price then
```

Comment:
Yep. @zathey can you make a new PR with his commit, then your suggested fix on top so we can get this moving along faster? Also, do you think there could be other related bugs? We should try to make sure our tests cover everything. Let me know if you know of any cases that are not covered by the tests.